active-rfcs/0000-function-api.md

```diff
@@ -136,7 +136,49 @@ const MyComponent = {
 
 Note this `props` object is reactive - i.e. it is updated when new props are passed in, and can be observed and reacted upon using the `watch` function introduced later in this RFC. However, for userland code, it is immutable during development (will emit warning if user code attempts to mutate it).
 
-`this` is usable inside `setup()`, but you most likely won't need it very often.
+The second argument provides a context object which exposes a number of properties that were previously exposed on `this` in 2.x APIs:
+
+``` js
+const MyComponent = {
+  setup(props, context) {
+    context.attrs
+    context.slots
+    context.refs
+    context.emit
```

Comment:
How can one listen to emitted events?

Reply:
`emit` here will be triggering listeners attached by the parent component.

It's no longer necessary to expose the full event emitter interface on the context - you can easily import and create an emitter instance inside `setup()`. We will likely provide a tree-shakable version of it in the standard build, and the full interface will still be available on `this` in the compatibility build.